torchci/pages/api/job_annotation/[repoOwner]/[repoName]/[annotation].ts

```diff
@@ -29,25 +29,36 @@ export default async function handler(
     return res.status(401).end();
   }
   const client = getDynamoClient();
-  const { jobId, repoOwner, repoName, annotation } = req.query;
-  const dynamoKey = `${repoOwner}/${repoName}/${jobId}`;
-
-  const item: any = {
-    dynamoKey,
-    repo: `${repoOwner}/${repoName}`,
-    jobID: parseInt(jobId as string),
-  };
-
-  // TODO: we encode annotations as a string, but probably we want to just
-  // serialize a JSON object instead to avoid this silly special case.
-  if (annotation !== "null") {
-    item["annotation"] = annotation;
-  }
+  const { repoOwner, repoName, annotation } = req.query;
+
+  // The request body contains an optional list of similar failures. If the list
+  // exists, the API will annotate all failures in the list with the same annotation
+  const jobIds = JSON.parse(req.body) ?? [];
```

Comment:
The new API endpoint here gets rid of the `jobId` in the URL and put them all into the request body